src/PatternBuilder.ts

```diff
@@ -166,29 +176,44 @@ export const between =
   })
 
 /** @since 1.0.0 */
-export const or =
-  (right: TermSequence | Atom | QuantifiedAtom) =>
-  (left: Pattern): Disjunction => ({
-    tag: 'disjunction',
-    left,
-    right,
-  })
+export const or: (
+  right: TermSequence | Atom | QuantifiedAtom
+) => (left: Pattern) => Disjunction = right => left => ({
+  tag: 'disjunction',
+  left,
+  right,
+})
+
+const getTerms: (termOrSeq: Term | TermSequence) => TermSequence['terms'] = match({
+  termSequence: ({ terms }) => terms,
+  atom: atom => [atom],
+  quantifiedAtom: qatom => [qatom],
+})
 
 /** @since 1.0.0 */
-export const then =
-  (term: Term) =>
-  (alt: TermSequence | Term): TermSequence => ({
-    tag: 'termSequence',
-    terms: alt.tag === 'termSequence' ? [...alt.terms, term] : [alt, term],
-  })
+export const then: (
+  term: Term | TermSequence
+) => (alt: TermSequence | Term) => TermSequence = term => alt => ({
+  tag: 'termSequence',
+  terms: [...getTerms(alt), ...getTerms(term)],
+})
```

Comment:
found a situation where I wanted to do `TermSequence, then TermSequence` and there's really no reason not to :woman_shrugging: